### PR TITLE
fix: JUnit processing for large files

### PR DESCRIPTION
**Pull request by devcatalin**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [x] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by vsukhin — APPROVED** [review]

File: cmd/testworkflow-toolkit/artifacts/walker.go [hunk 1]
```diff
@@ -197,6 +197,8 @@ func (w *walker) walk(fsys fs.FS, path string, walker WalkerFn) error {
 		sanitizedPath = "."
 	}
 
+	fmt.Printf("Walking: %s\n", sanitizedPath)
```
Comment (rangoo94) [inline cmd/testworkflow-toolkit/artifacts/walker.go:200]: Likely debug log

File: cmd/testworkflow-toolkit/artifacts/walker.go [hunk 2]
```diff
@@ -219,6 +221,7 @@ func (w *walker) walk(fsys fs.FS, path string, walker WalkerFn) error {
 
 		// Pass the data to final walker
 		relativeFilePath := strings.TrimLeft(resolvedPath[len(w.root):], "/")
+		fmt.Printf("Relative path: %s\n", relativeFilePath)
```
Comment (rangoo94) [inline cmd/testworkflow-toolkit/artifacts/walker.go:224]: Likely debug log

File: cmd/testworkflow-toolkit/commands/artifacts.go [hunk 3]
```diff
@@ -180,6 +180,8 @@ func run(handler artifacts.Handler, walker artifacts.Walker, dirFS fs.FS) {
 
 	started := time.Now()
 	err = walker.Walk(dirFS, func(path string, file fs.File, _ fs.FileInfo, err error) error {
+		fmt.Printf("Processing: %s\n", ui.LightCyan(path))
```
Comment (rangoo94) [inline cmd/testworkflow-toolkit/commands/artifacts.go:183]: Likely debug log

File: cmd/testworkflow-toolkit/artifacts/junit_post_processor.go [hunk 4]
```diff
@@ -63,10 +62,7 @@ func (p *JUnitPostProcessor) Add(path string) error {
 	if err != nil {
 		return errors.Wrapf(err, "failed to read %s", path)
 	}
-	ok, err := isJUnitReport(xmlData)
-	if err != nil {
-		return errors.Wrapf(err, "failed to check if %s is a JUnit report", path)
-	}
+	ok := isJUnitReport(xmlData)
```
Comment (rangoo94) [inline cmd/testworkflow-toolkit/artifacts/junit_post_processor.go:65]: Probably best to not load the whole `xmlData` then - just the first 8KB as below

**Review by rangoo94 — COMMENTED** [review]
Seems like there are some debugging logs for removal, unnecessary memory storage for (big) files check

**Pull request merged**